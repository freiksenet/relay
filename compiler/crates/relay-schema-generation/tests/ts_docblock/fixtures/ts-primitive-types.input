/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

//- module.ts

import type CatFlowType from 'Cat';

/**
 * @RelayResolver
 */
export function Cat(id: DataID): CatFlowType {
  return {};
}


/**
 * @RelayResolver
 */
export function ages(cat: CatFlowType): number | null {
  return 2;
}


/**
 * @RelayResolver
 */
export function hungry1(cat: CatFlowType): boolean {
  return false;
}

/**
 * @RelayResolver
 */
export function hungry2(cat: CatFlowType): false | null | undefined {
  return false;
}

/**
 * @RelayResolver
 */
export function hungry3(cat: CatFlowType): true | undefined {
  return false;
}
