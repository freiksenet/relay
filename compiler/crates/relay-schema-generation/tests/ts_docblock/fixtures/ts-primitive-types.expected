==================================== INPUT ====================================
/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

//- module.ts

import type CatFlowType from 'Cat';

/**
 * @RelayResolver
 */
export function Cat(id: DataID): CatFlowType {
  return {};
}


/**
 * @RelayResolver
 */
export function ages(cat: CatFlowType): number | null {
  return 2;
}


/**
 * @RelayResolver
 */
export function hungry1(cat: CatFlowType): boolean {
  return false;
}

/**
 * @RelayResolver
 */
export function hungry2(cat: CatFlowType): false | null | undefined {
  return false;
}

/**
 * @RelayResolver
 */
export function hungry3(cat: CatFlowType): true | undefined {
  return false;
}
==================================== OUTPUT ===================================
Field(
    TerseRelayResolver(
        TerseRelayResolverIr {
            field: FieldDefinition {
                name: Identifier {
                    span: 362:366,
                    token: Token {
                        span: 362:366,
                        kind: Identifier,
                    },
                    value: "ages",
                },
                type_: Named(
                    NamedTypeAnnotation {
                        name: Identifier {
                            span: 386:392,
                            token: Token {
                                span: 386:392,
                                kind: Identifier,
                            },
                            value: "Float",
                        },
                    },
                ),
                arguments: None,
                directives: [],
                description: None,
                hack_source: None,
                span: 362:366,
            },
            type_: WithLocation {
                location: module.ts:272:275,
                item: "Cat",
            },
            root_fragment: None,
            deprecated: None,
            semantic_non_null: None,
            live: None,
            location: module.ts:362:366,
            fragment_arguments: None,
            source_hash: ResolverSourceHash(
                "793f10cbcaa057bbffad9b50e62171c7",
            ),
        },
    ),
)
extend type Cat {
  ages: Float @relay_resolver(fragment_name: "Cat____relay_model_instance", generated_fragment: true, inject_fragment_data: "__relay_model_instance", has_output_type: true, import_name: "ages", import_path: "module.ts") @resolver_source_hash(value: "793f10cbcaa057bbffad9b50e62171c7")
}


Field(
    TerseRelayResolver(
        TerseRelayResolverIr {
            field: FieldDefinition {
                name: Identifier {
                    span: 460:467,
                    token: Token {
                        span: 460:467,
                        kind: Identifier,
                    },
                    value: "hungry1",
                },
                type_: Named(
                    NamedTypeAnnotation {
                        name: Identifier {
                            span: 487:494,
                            token: Token {
                                span: 487:494,
                                kind: Identifier,
                            },
                            value: "Boolean",
                        },
                    },
                ),
                arguments: None,
                directives: [],
                description: None,
                hack_source: None,
                span: 460:467,
            },
            type_: WithLocation {
                location: module.ts:272:275,
                item: "Cat",
            },
            root_fragment: None,
            deprecated: None,
            semantic_non_null: Some(
                ConstantDirective {
                    span: 460:467,
                    at: Token {
                        span: 0:0,
                        kind: Empty,
                    },
                    name: Identifier {
                        span: 460:467,
                        token: Token {
                            span: 0:0,
                            kind: Empty,
                        },
                        value: "semanticNonNull",
                    },
                    arguments: None,
                },
            ),
            live: None,
            location: module.ts:460:467,
            fragment_arguments: None,
            source_hash: ResolverSourceHash(
                "793f10cbcaa057bbffad9b50e62171c7",
            ),
        },
    ),
)
extend type Cat {
  hungry1: Boolean @relay_resolver(fragment_name: "Cat____relay_model_instance", generated_fragment: true, inject_fragment_data: "__relay_model_instance", has_output_type: true, import_name: "hungry1", import_path: "module.ts") @resolver_source_hash(value: "793f10cbcaa057bbffad9b50e62171c7") @semanticNonNull
}


Field(
    TerseRelayResolver(
        TerseRelayResolverIr {
            field: FieldDefinition {
                name: Identifier {
                    span: 558:565,
                    token: Token {
                        span: 558:565,
                        kind: Identifier,
                    },
                    value: "hungry2",
                },
                type_: Named(
                    NamedTypeAnnotation {
                        name: Identifier {
                            span: 585:590,
                            token: Token {
                                span: 585:590,
                                kind: Identifier,
                            },
                            value: "Boolean",
                        },
                    },
                ),
                arguments: None,
                directives: [],
                description: None,
                hack_source: None,
                span: 558:565,
            },
            type_: WithLocation {
                location: module.ts:272:275,
                item: "Cat",
            },
            root_fragment: None,
            deprecated: None,
            semantic_non_null: None,
            live: None,
            location: module.ts:558:565,
            fragment_arguments: None,
            source_hash: ResolverSourceHash(
                "793f10cbcaa057bbffad9b50e62171c7",
            ),
        },
    ),
)
extend type Cat {
  hungry2: Boolean @relay_resolver(fragment_name: "Cat____relay_model_instance", generated_fragment: true, inject_fragment_data: "__relay_model_instance", has_output_type: true, import_name: "hungry2", import_path: "module.ts") @resolver_source_hash(value: "793f10cbcaa057bbffad9b50e62171c7")
}


Field(
    TerseRelayResolver(
        TerseRelayResolverIr {
            field: FieldDefinition {
                name: Identifier {
                    span: 673:680,
                    token: Token {
                        span: 673:680,
                        kind: Identifier,
                    },
                    value: "hungry3",
                },
                type_: Named(
                    NamedTypeAnnotation {
                        name: Identifier {
                            span: 700:704,
                            token: Token {
                                span: 700:704,
                                kind: Identifier,
                            },
                            value: "Boolean",
                        },
                    },
                ),
                arguments: None,
                directives: [],
                description: None,
                hack_source: None,
                span: 673:680,
            },
            type_: WithLocation {
                location: module.ts:272:275,
                item: "Cat",
            },
            root_fragment: None,
            deprecated: None,
            semantic_non_null: None,
            live: None,
            location: module.ts:673:680,
            fragment_arguments: None,
            source_hash: ResolverSourceHash(
                "793f10cbcaa057bbffad9b50e62171c7",
            ),
        },
    ),
)
extend type Cat {
  hungry3: Boolean @relay_resolver(fragment_name: "Cat____relay_model_instance", generated_fragment: true, inject_fragment_data: "__relay_model_instance", has_output_type: true, import_name: "hungry3", import_path: "module.ts") @resolver_source_hash(value: "793f10cbcaa057bbffad9b50e62171c7")
}


Type(
    StrongObjectResolver(
        StrongObjectIr {
            type_name: Identifier {
                span: 272:275,
                token: Token {
                    span: 272:275,
                    kind: Identifier,
                },
                value: "Cat",
            },
            rhs_location: module.ts:272:275,
            root_fragment: WithLocation {
                location: module.ts:272:275,
                item: FragmentDefinitionName(
                    "Cat__id",
                ),
            },
            description: None,
            deprecated: None,
            live: None,
            semantic_non_null: None,
            location: module.ts:272:275,
            implements_interfaces: [],
            source_hash: ResolverSourceHash(
                "793f10cbcaa057bbffad9b50e62171c7",
            ),
        },
    ),
)
type Cat @__RelayResolverModel {
  id: ID!
  __relay_model_instance: RelayResolverValue! @relay_resolver(generated_fragment: true, fragment_name: "Cat__id", import_name: "Cat", import_path: "module.ts", inject_fragment_data: "id") @resolver_source_hash(value: "793f10cbcaa057bbffad9b50e62171c7") @unselectable(reason: "This field is intended only for Relay's internal use")
}
